File: client/src/__tests__/selenium/admin_test.js
const { By, Key, Builder, until } = require("selenium-webdriver");
require("chromedriver");
BASE_URL = "http://localhost:5173/admin"
async function adminDashboardTest() {
    // Start browser
    let driver = await new Builder().forBrowser("chrome").build();

    try {
        // Navigate to Admin Dashboard
        await driver.get(BASE_URL); // Change to your actual local or deployed URL
        await driver.manage().window().setWindowSize(1024, 768);
        // Wait until dashboard header appears
        await driver.wait(
            until.elementLocated(By.xpath("//*[contains(text(),'📊 Admin Dashboard Overview')]")),
            10000
        );
        console.log("✅ Admin Dashboard loaded successfully.");

        // Verify Summary section exists
        await driver.wait(
            until.elementLocated(By.xpath("//*[contains(text(),'Summary (This Month)')]")),
            5000
        );
        console.log("✅ Summary section is visible.");

        // Verify Sales Over Time chart is present
        await driver.wait(
            until.elementLocated(By.xpath("//*[contains(text(),'Sales Over Time')]")),
            5000
        );
        console.log("✅ Sales Over Time chart is visible.");

        // Verify Revenue Over Time chart is present
        await driver.wait(
            until.elementLocated(By.xpath("//*[contains(text(),'Revenue Over Time')]")),
            5000
        );
        console.log("✅ Revenue Over Time chart is visible.");

        // Click the Download Detailed Report button
        let downloadButton = await driver.findElement(
            By.xpath("//button[contains(text(),'Download Detailed Report')]")
        );
        await downloadButton.click();
        console.log("✅ Download Detailed Report button clicked.");

        // Verify Top 10 Best-Selling Products table
        await driver.wait(
            until.elementLocated(By.xpath("//*[contains(text(),'Top 10 Best-Selling Products')]")),
            5000
        );
        console.log("✅ Top 10 Best-Selling Products table is visible.");

    } catch (err) {
        console.error("❌ Test failed:", err);
    } finally {
        // Close browser
        await driver.quit();
    }
}

adminDashboardTest();

File: client/src/__tests__/selenium/admin_test.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

# --- Configuration ---
BASE_URL = "http://localhost:5173/admin"  # Change to your actual Admin Dashboard URL

# --- Setup ---


service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)
wait = WebDriverWait(driver, 10)

def test_google_search(driver: webdriver.Chrome):
    driver.get("https://www.google.com")
    assert "Google" in driver.title
    print("Google search page loaded successfully.")

def test_admin_dashboard(driver: webdriver.Chrome):
    driver.get(BASE_URL)
    print("✅ Admin Dashboard page loaded successfully.")

    wait.until(EC.presence_of_element_located(
        (By.XPATH, "//*[contains(text(),'📊 Admin Dashboard Overview')]")
    ))
    print("✅ Admin Dashboard loaded successfully.")

    wait.until(EC.presence_of_element_located(
        (By.XPATH, "//*[contains(text(),'Summary (This Month)')]")
    ))
    print("✅ Summary section visible.")

    wait.until(EC.presence_of_element_located(
        (By.XPATH, "//*[contains(text(),'Sales Over Time')]")
    ))
    print("✅ Sales Over Time chart visible.")

    wait.until(EC.presence_of_element_located(
        (By.XPATH, "//*[contains(text(),'Revenue Over Time')]")
    ))
    print("✅ Revenue Over Time chart visible.")

    download_button = wait.until(EC.element_to_be_clickable(
        (By.XPATH, "//button[contains(text(),'Download Detailed Report')]")
    ))
    download_button.click()
    print("✅ Download Detailed Report button clicked.")

    wait.until(EC.presence_of_element_located(
        (By.XPATH, "//*[contains(text(),'Top 10 Best-Selling Products')]")
    ))
    print("✅ Top 10 Best-Selling Products table visible.")

try:
    test_google_search(driver)
    test_admin_dashboard(driver)

except Exception as e:
    print("❌ Test failed:", e)

finally:
    driver.quit()
